aspnetcore/security/authorization/secure-data/samples/final15/Authorization/ContactRoleAuthorizationHandler.cs

```diff
@@ -16,7 +16,7 @@ public class ContactRoleAuthorizationHandler
                 return Task.FromResult(0);
             }
 
-            if (context.User.IsInRole(requirement.Name))
+            if (context.User.IsInRole(Constants.OperationRolePrefix + requirement.Name))
```

Comment:
I guess it isn't quite clear to me what we are trying to illustrate with this sample. If it is to show how to do granular control of authorization over operations, then what would be the realistic scenario, a role or claim or policy per operation? I guess it would seem strange to use a single role or policy that grants all of the operations because if that is sufficient then we probably wouldn't use operation level authorization. Or does that matter, are we just trying to show the mechanics of it or a realistic scenario of when you might use it? I'm happy to help with this but I'm not the subject matter expert, I'm learning this stuff from you guys, so any clarity or guidance you could provide would be appreciated.